packages/store-cache/src/utils.ts

```diff
@@ -201,18 +201,34 @@ function tupleToRecord(tuple: Tuple): Record<string, any> {
     // Ignore all non-object tuple values
     if (entry === null || Array.isArray(entry) || typeof entry !== "object") continue;
     for (const [key, value] of Object.entries(entry)) {
-      record[key] = value;
+      record[key] = deserializeKey(value);
     }
   }
   return record;
 }
 
 /**
  * Helper to serialize values that are not natively supported in keys by tuple-database.
+ * (see https://github.com/ccorcos/tuple-database/issues/25)
  * For now only `bigint` needs serialization.
  */
 function serializeKey(key: unknown) {
-  if (typeof key === "bigint") return String(key);
+  if (typeof key === "bigint") return `${key.toString()}n`;
```

Comment:
do you know if there are any pending specs for serializing bigints? thinking about JSON etc. where it may not have landed but maybe coming?

ultimately wondering if the `n` suffix is aligned with whatever the JS ecosystem is thinking as the standard for encoding/serializing these kinds of values (vs a plain number string, but just much larger than what JS can support in terms of a number)

and brings up another question in my mind around caching - do we have or will we have a "version" stamp for a given store cache so that, if we need to change the internal serialization, we can bump the version to "invalidate" the cache and get clients to rebuild from scratch with the new serialization method?

Reply:
> do you know if there are any pending specs for serializing bigints? thinking about JSON etc. where it may not have landed but maybe coming?

From https://github.com/GoogleChromeLabs/jsbi/issues/30: 
> Working as intended/spec'ed: https://tc39.es/proposal-bigint/#sec-serializejsonproperty (step 10).
>
> In short, the background is that JSON is so commonly used among all sorts of different systems and programming languages that it is effectively impossible to add anything to the format without breaking compatibility in some crucial case.

So it seems like won't be support for `bigint` in JSON and it doesn't seem like there is a standard in progress.

> and brings up another question in my mind around caching - do we have or will we have a "version" stamp for a given store cache so that, if we need to change the internal serialization, we can bump the version to "invalidate" the cache and get clients to rebuild from scratch with the new serialization method?

Agree with versioning! In the current version `store-cache` doesn't persist the data anywhere yet, but that's definitely planned. I'd probably add the versioning in the PR where we add functionality to persist the cache so it can be informed by the needs we have there.